### PR TITLE
Bump go-containerregistry to latest

### PR DESCRIPTION
**Pull request by sayboras**
fixes https://github.com/kubernetes/minikube/issues/6045

### CODE REVIEW

**Comment by tstromberg** [comment]
Does `make` work for you? I noticed the build failed on Travis:

```
# github.com/google/go-containerregistry/pkg/v1/daemon
../../../../pkg/mod/github.com/google/go-containerregistry@v0.0.0-20191211193041-0eaa33c3d13c/pkg/v1/daemon/image.go:73:14: undefined: client.NewClientWithOpts
../../../../pkg/mod/github.com/google/go-containerregistry@v0.0.0-20191211193041-0eaa33c3d13c/pkg/v1/daemon/image.go:73:39: undefined: client.FromEnv
../../../../pkg/mod/github.com/google/go-containerregistry@v0.0.0-20191211193041-0eaa33c3d13c/pkg/v1/daemon/write.go:38:14: undefined: client.NewClientWithOpts
../../../../pkg/mod/github.com/google/go-containerregistry@v0.0.0-20191211193041-0eaa33c3d13c/pkg/v1/daemon/write.go:38:39: undefined: client.FromEnv
Makefile:149: recipe for target 'out/minikube' failed
make: *** [out/minikube] Error 2
The command "make" exited with 2.
```

**Comment by sayboras** [comment]
@tstromberg The latest version of go-containerregistry requires docker/docker github.com/docker/docker v1.4.2-0.20190924003213-a8608b5b67c7. However, this version will break https://github.com/samalba/dockerclient (for test cases only). As this client is no longer updated and maintained (https://github.com/docker/swarm/issues/1879). For the time being, I have raised https://github.com/machine-drivers/machine/issues/26 to replace it.

Need your input if there is any alternative approach.

```
= go mod ================================================================
k8s.io/minikube/pkg/minikube/cluster imports
        github.com/docker/machine/libmachine/provision imports
        github.com/samalba/dockerclient tested by
        github.com/samalba/dockerclient.test imports
        github.com/docker/docker/pkg/jsonlog: module github.com/docker/docker@latest found (v1.13.1, replaced by github.com/docker/docker@v1.4.2-0.20190924003213-a8608b5b67c7), but does not contain package github.com/docker/docker/pkg/jsonlog
```

**Comment by sayboras** [comment]
wait for https://github.com/samalba/dockerclient/pull/246 to address the above issue, not sure if the author is still checking PR.

**Comment by medyagh** [comment]
@sayboras  are we still waiting on the downstream PR before we merge this ? or is this good to go?

**Comment by medyagh** [comment]
@sayboras the cache reload test fails consitantly across different drivers


```
=== RUN   TestFunctional/serial/CacheCmd/cache/cache_reload
--- FAIL: TestFunctional/serial/CacheCmd/cache/cache_reload (6.09s)
functional_test.go:360: (dbg) Run:  out/minikube-linux-amd64 -p functional-20200107T141113.44282597-5018 ssh sudo docker rmi busybox:latest
functional_test.go:360: (dbg) Non-zero exit: out/minikube-linux-amd64 -p functional-20200107T141113.44282597-5018 ssh sudo docker rmi busybox:latest: exit status 1 (1.094145272s)
-- stdout --
Error: No such image: busybox:latest
-- /stdout --
** stderr ** 
ssh: Process exited with status 1
** /stderr **
functional_test.go:362: failed to delete inside the node "out/minikube-linux-amd64 -p functional-20200107T141113.44282597-5018 ssh sudo docker rmi busybox:latest" : exit status 1
functional_test.go:365: (dbg) Run:  out/minikube-linux-amd64 -p functional-20200107T141113.44282597-5018 ssh sudo crictl inspecti busybox:latest
functional_test.go:365: (dbg) Non-zero exit: out/minikube-linux-amd64 -p functional-20200107T141113.44282597-5018 ssh sudo crictl inspecti busybox:latest: exit status 1 (1.012306754s)
-- stdout --
[31mFATA[0m[0000] no such image "busybox:latest" present       
-- /stdout --
** stderr ** 
ssh: Process exited with status 1
** /stderr **
functional_test.go:370: (dbg) Run:  out/minikube-linux-amd64 -p functional-20200107T141113.44282597-5018 cache reload
functional_test.go:370: (dbg) Done: out/minikube-linux-amd64 -p functional-20200107T141113.44282597-5018 cache reload: (3.428518998s)
functional_test.go:375: (dbg) Run:  out/minikube-linux-amd64 -p functional-20200107T141113.44282597-5018 ssh sudo crictl inspecti busybox:latest
functional_test.go:375: (dbg) Non-zero exit: out/minikube-linux-amd64 -p functional-20200107T141113.44282597-5018 ssh sudo crictl inspecti busybox:latest: exit status 1 (559.672603ms)
-- stdout --
[31mFATA[0m[0000] no such image "busybox:latest" present       
-- /stdout --
** stderr ** 
ssh: Process exited with status 1
** /stderr **
functional_test.go:377: expected to get no error for "out/minikube-linux-amd64 -p functional-20200107T141113.44282597-5018 ssh sudo crictl inspecti busybox:latest" but got exit status 1
```

**Comment by sayboras** [comment]
@medyagh oh sorry about this, let me try to debug it

**Comment by medyagh** [comment]
@sayboras are you still working on this ?

**Comment by sayboras** [comment]
> @sayboras are you still working on this ?

yes, I managed to replicate the issue in local development env (as the pipeline is running quite long), hopefully I sort it out next few days.

**Comment by medyagh** [comment]
@sayboras are you still working on this ? I would be happy to see this PR merged once integration tests and conflicts are solved.

**Comment by sayboras** [comment]
@medyagh need your help on integration test, I tried to run it locally on my macbook, but got another issue :(. I have ubuntu and mac for testing. Appreciate if you can give me how to do it so that feedback is faster.

FYI. I compile minikube binary and manage to provision cluster successfully.

**Comment by priyawadhwa** [comment]
Hey @sayboras thank you for opening this PR. I think it's pretty close to being merged, could you please resolve merge conflicts?

Also, what issue did you run into when running integration tests locally? We have some [documentation](https://github.com/kubernetes/minikube/tree/master/test/integration) around it which could be helpful.

**Comment by medyagh** [comment]
@sayboras  are you still interested to finish this PR, seems like only needs to resolve merge conflicts?

**Comment by sayboras** [comment]
> @sayboras are you still interested to finish this PR, seems like only needs to resolve merge conflicts?

I just rebased with master, however, some of the test are failing still, cross checking with other PR, seems like they are having the same issue as well. Few are related to busybox like https://storage.googleapis.com/minikube-builds/logs/6073/Docker_Linux.html#failtestwindow2

File: test/integration/functional_test.go [hunk 1]
```diff
@@ -334,7 +334,7 @@ func validateCacheCmd(ctx context.Context, t *testing.T, profile string) {
 	}
 	t.Run("cache", func(t *testing.T) {
 		t.Run("add", func(t *testing.T) {
-			for _, img := range []string{"busybox", "busybox:1.28.4-glibc", "k8s.gcr.io/pause:latest"} {
+			for _, img := range []string{"busybox:latest", "busybox:1.28.4-glibc", "k8s.gcr.io/pause:latest"} {
```
Comment (sayboras) [inline test/integration/functional_test.go]: we need to have `latest` due to https://github.com/google/go-containerregistry/pull/404/files, which allows to have both tag and digest, hence the `latest` tag is not added automatically.

**Comment by sayboras** [comment]
> @sayboras are you still interested to finish this PR, seems like only needs to resolve merge conflicts?

Oh sorry, just have some time locked down to finish this. It should be ok now, please let me know if you have any feedback, happy to address.

**Comment by medyagh** [comment]
@sayboras  thanks for updating the PR, the Hyperkit Test failures are a lot !

I wonder if this is related to this PR ? https://storage.googleapis.com/minikube-builds/logs/6073/HyperKit_macOS.html


have you tried it on mac yourself? to run the tests 

you can run the tests using this command:
if you have a running minikube profile u can test against it this way.

```
make integration -e TEST_ARGS="-test.run TestFunctional --profile=minikube --cleanup=false"
```

**Comment by sayboras** [comment]
> @sayboras thanks for updating the PR, the Hyperkit Test failures are a lot !
> 
> I wonder if this is related to this PR ? https://storage.googleapis.com/minikube-builds/logs/6073/HyperKit_macOS.html
> 
> have you tried it on mac yourself? to run the tests
> 
> you can run the tests using this command:
> if you have a running minikube profile u can test against it this way.
> 
> ```
> make integration -e TEST_ARGS="-test.run TestFunctional --profile=minikube --cleanup=false"
> ```

Yes, I did with with the cleanup=true

```code
 make integration -e TEST_ARGS="-test.run TestFunctional --profile=minikube --cleanup=true"  
....
PASS
ok      k8s.io/minikube/test/integration        168.192s
```

**Comment by medyagh** [comment]
Thank you very much :)

**Pull request merged**